### PR TITLE
Coverity 1022104 Dereference before null check

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

File: proxy/http/remap/RemapPlugins.cc [hunk 1]
```diff
@@ -28,11 +28,14 @@ ClassAllocator<RemapPlugins> pluginAllocator("RemapPluginsAlloc");
 TSRemapStatus
 RemapPlugins::run_plugin(remap_plugin_info *plugin)
 {
+  ink_assert(_s);
```
Comment (PSUdaemon) [inline proxy/http/remap/RemapPlugins.cc]: Thoughts on making this release assert?

Comment (jpeach) [inline proxy/http/remap/RemapPlugins.cc]: Yes a release assert would be justified here.

**Comment by jpeach** [comment]
@zwoop a couple of additional suggestions

- rename `_s` to something more meaningful
- replace the 4 `setXxx()` member functions with a single `init()`

**Comment by zwoop** [comment]
We can make it a release assert (although I'm hesitant to do that since it never happens so it's an additional, unnecessary check). The other suggestions, feel free to fix in subsequent PRs.

**Pull request merged**